sale_crm_event_reservation/wizards/crm_lead_event_sale_wizard.py

```diff
@@ -41,60 +41,68 @@ class CRMLeadEventSale(models.TransientModel):
                 ("event_reservation_type_id", "=", event_type_id),
             ]
         """,
-        context="""
-            {
-                "default_event_reservation_ok": True,
-                "default_event_reservation_type_id": event_type_id,
-            }
-        """,
         index=True,
         ondelete="cascade",
         string="Product",
     )
+    allowed_event_ids = fields.Many2many(
+        comodel_name="event.event",
+        compute="_compute_allowed_event_ids",
+        readonly=True,
+    )
     event_id = fields.Many2one(
         comodel_name="event.event",
-        domain="""
-            [
-                ("event_type_id", "=", event_type_id),
-                ("date_end", ">=", datetime.date.today().strftime("%Y-%m-%d")),
-                ("is_finished", "!=", True),
-                "|",
-                ("seats_limited", "=", False),
-                ("seats_available", ">=", seats_wanted),
-            ]
-        """,
-        context="""
-            {
-                "default_event_type_id": event_type_id,
-            }
-        """,
+        domain="[('id', 'in', allowed_event_ids)]",
         index=True,
         ondelete="cascade",
         string="Event",
     )
+    allowed_event_ticket_ids = fields.Many2many(
+        comodel_name="event.event.ticket",
+        compute="_compute_allowed_event_ticket_ids",
+        readonly=True,
+    )
     event_ticket_id = fields.Many2one(
         comodel_name="event.event.ticket",
-        domain="""
-            [
-                ("event_id", "=", event_id),
-                "|",
-                ("end_sale_datetime", "=", False),
-                ("end_sale_datetime", ">=", datetime.date.today().strftime("%Y-%m-%d")),
-                "|",
-                ("seats_limited", "=", False),
-                ("seats_available", ">=", seats_wanted),
-            ]
-        """,
-        context="""
-            {
-                "default_event_id": event_id,
-            }
-        """,
+        domain="[('id', 'in', allowed_event_ticket_ids)]",
         index=True,
         ondelete="cascade",
         string="Ticket",
     )
 
+    @api.depends("mode")
+    def _compute_allowed_event_ids(self):
+        for record in self:
+            events = self.env["event.event"].search(
```

Comment:
I'm quite doubtful on this...

Reply:
Well, event_event isn't usually a massive table (our most intesive use customer has some thousands over more than 10 years). I don't see a performance issue... and this is just a wizard